lib/utils.ts

```diff
@@ -395,6 +408,36 @@ async function requestAsync(
 		opts.headers = nativeHeaders;
 	}
 
+	if (hasBlob(options.body)) {
```

Comment:
I guess we can avoid iterating over the body props twice.
Just a single `for (const [k, v] of Object.entries(options.body)) {` that switches the request to multipart on the first occurrence of a Blob should be enough.

Reply:
@thgreasi actually I am not sure we can avoid it for the NodeFormData. We can remove the hasBlob loop indeed, but in case of it being a node form data we require the second pass as `form.getHeaders()` will have calculated the boundary (which is not necessary for browsers as we can just remove the Content-Type header and the browser can calculate the boundary itself before sending the request)